### PR TITLE
[Sema] Unify fix-it logic for enclosing trailing closure in argument parens

### PR DESCRIPTION
**Pull request by rintaro**
`tryDiagnoseTrailingClosureAmbiguity` and `checkStmtConditionTrailingClosure`
had similar logic.
Added `swift::fixItEncloseTrailingClosure` function.

### CODE REVIEW

**Comment by rintaro** [comment]
@jrose-apple Could you review this?
Although this is slightly inefficient than before, I think that it's not a problem.

File: lib/Sema/MiscDiagnostics.cpp [hunk 1]
```diff
@@ -3098,6 +3098,51 @@ static void checkSwitch(TypeChecker &TC, const SwitchStmt *stmt) {
   }
 }
 
+void swift::fixItEncloseTrailingClosure(TypeChecker &TC,
+                                        InFlightDiagnostic &diag,
+                                        const CallExpr *call,
+                                        Identifier closureLabel) {
+  auto argsExpr = call->getArg();
+  if (auto TSE = dyn_cast<TupleShuffleExpr>(argsExpr))
+    argsExpr = TSE->getSubExpr();
```
Comment (jrose-apple) [inline lib/Sema/MiscDiagnostics.cpp]: I forgot this one in my new code, too. >< Thanks.

**Review by jrose-apple — APPROVED** [review]
I should have realized this code already existed. Thanks, Rintaro!

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**